playwright.config.ts

```diff
@@ -0,0 +1,45 @@
+import { PlaywrightTestConfig, devices } from "@playwright/test";
+
+const opts = {
+  // launch headless on CI, in browser locally
+  headless: !!process.env.CI || !!process.env.PLAYWRIGHT_HEADLESS,
+  collectCoverage: false, // not possible in Next.js 12
+  executablePath: process.env.PLAYWRIGHT_CHROME_EXECUTABLE_PATH,
+};
+
+console.log("⚙️ Playwright options:", JSON.stringify(opts, null, 4));
+
+const config: PlaywrightTestConfig = {
+  forbidOnly: !!process.env.CI,
+  testDir: "playwright",
+  timeout: 60e3,
+  retries: process.env.CI ? 3 : 0,
+  globalSetup: require.resolve("./playwright/lib/globalSetup"),
+  use: {
+    baseURL: "http://localhost:3000",
+    locale: "en-US",
+    trace: "on-first-retry",
+    headless: opts.headless,
+    contextOptions: {
```

Comment:
Instead of doing it via the context options, do it via the test-runner itself, set the strategy to e.g. "on-first-retry" and benefit from huge perf improvements. (There you can set then a custom outputDir).

Also we usually recommended tracing over video, since it gives you more, e.g. inspect the DOM, see Playwright actions etc.
https://playwright.dev/docs/test-configuration#record-test-trace

Reply:
So would you recommend moving the trace settings to per test setting? Or I'm misunderstanding

Reply:
I'm recommending to not use - contextOptions and use video: "on-first-retry" instead see here: https://playwright.dev/docs/test-configuration#record-video

Having video always enabled is CPU heavy etc. Why use tracing instead? Because they are way better for debugging actual failures (show DOM, networking, console, Playwright steps etc).

Reply:
Gotcha, will adjust on a follow PR, thanks again @mxschmitt 🙌🏽

Reply:
Also, should I leave contextOptions to keep the videos in that directory? I couldn't find another way to specify an output dir for videos.